### PR TITLE
Support transactions

### PR DESCRIPTION
**Pull request by YordanPavlov**
This PR adds Kafka transaction support to san-exporter. We need this to provide 'exactly once' semantic.

The bigger work related with this PR has been happening in Node-rdkafka. The library is a wrapper over Librdkafka which supports transactions, but that support needs to be brought to JS. I have been trying to do this in this PR:
https://github.com/Blizzard/node-rdkafka/pull/881
If you feel like doing so, please also comment there. My PR is still not accepted so for now we use an NPM from our own NPM repo: `@santiment-network/node-rdkafka`.

Also in this PR I have done some fixes:
* Zookeeper client is now a member instead of global variable. Global variable was causing weird problems in integration tests, when the san-exporter is dropped and created again.
* I have added an example for using transactions and also made the examples runnable as a sort of test.
* Added a helper method to subscribe for message delivery.

P.S. I am tagging more people for this PR, because we have bigger plans for san-exporter. The idea is for it to gradually absorb all other exporters into a single higher quality repository which people outside of Santiment would use. So any feedback is welcome.

### CODE REVIEW

**Review by apozdeyev — APPROVED** [review]
LGTM

**Review by kalkov — APPROVED** [review]

File: examples/send_dates_transaction.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const pkg = require('../package.json');
+const { Exporter } = require('../index')
+const exporter = new Exporter(pkg.name, true)
+
+async function pushData() {
+  const timestamp = Math.floor(new Date() / 1000)
```
Comment (IvanIvanoff) [inline examples/send_dates_transaction.js:6]: nitpick: I'm not sure how the implicit type conversion is handled here. If you want to not use type coercion you can do: `Math.floor(new Date().getTime() / 1000)`

File: examples/send_dates_transaction.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  let lastPosition = await exporter.getLastPosition()
+  console.log(`Last position: ${JSON.stringify(lastPosition)}`)
+  let key = 1
+  if(lastPosition) {
+    key = lastPosition.key + 1
+  }
```
Comment (IvanIvanoff) [inline examples/send_dates_transaction.js:13]: You can make use of the ternary operator to avoid double assignments:
```js
let key = lastPosition ? lastPosition.key +1 : 1
```

File: examples/send_dates_transaction.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  console.log("Sending data with transaction...")
+  await exporter.sendDataWithKey({
+    timestamp: timestamp,
+    iso_date: new Date().toISOString(),
```
Comment (IvanIvanoff) [inline examples/send_dates_transaction.js:18]: Fetching of the last position does take some time, so these 2 will end up pointing to different times. I suggest you have `const now = new Date()` at the top and use it both for the timestamp and iso_date

File: examples/send_dates_transaction.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    key: key
+  }, "key")
+
+  let position = {timestamp: timestamp, key: key}
+  console.log(`Saving position: ${JSON.stringify(position)}`)
+  await exporter.savePosition(position)
+
+  let newPosition = await exporter.getLastPosition()
+  console.log(`New position: ${JSON.stringify(newPosition)}`)
```
Comment (IvanIvanoff) [inline examples/send_dates_transaction.js:27]: Why is this needed? If there is possible concurrent usage, I think you would want to store the position which you just saved instead of the one returned by `getLastPosition`. If there are no race conditions - can't you just print `position`?

File: examples/send_dates_transaction.js (hunk 1 above)
Comment (YordanPavlov) [reply examples/send_dates_transaction.js]: Good point, changed.

File: examples/send_dates_transaction.js (hunk 2 above)
Comment (YordanPavlov) [reply examples/send_dates_transaction.js]: Yes, much more readable.

File: examples/send_dates_transaction.js (hunk 3 above)
Comment (YordanPavlov) [reply examples/send_dates_transaction.js]: Yes, storing exactly the same time better illustrate the purpose of the example, changed.

File: examples/send_dates_transaction.js (hunk 4 above)
Comment (YordanPavlov) [reply examples/send_dates_transaction.js:24]: I believe the purpose of the example is to illustrate that the 'position' has been successfully stored in Zookeeper.

**Review by IvanIvanoff — APPROVED** [review]
